devicetypes/stelpro/stelpro-ki-zigbee-thermostat.src/stelpro-ki-zigbee-thermostat.groovy

```diff
@@ -129,28 +129,28 @@ def setupHealthCheck() {
 	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
 }
 
-def installed() {
-	setupHealthCheck()
-
+def configureSupportedRanges() {
 	sendEvent(name: "supportedThermostatModes", value: supportedThermostatModes, displayed: false)
 	sendEvent(name: "thermostatSetpointRange", value: thermostatSetpointRange, displayed: false)
 	sendEvent(name: "heatingSetpointRange", value: heatingSetpointRange, displayed: false)
 }
 
+def installed() {
+	setupHealthCheck()
+
+	configureSupportedRanges()
+}
+
 def updated() {
 	def requests = []
 	setupHealthCheck()
 
-	sendEvent(name: "supportedThermostatModes", value: supportedThermostatModes, displayed: false)
-	sendEvent(name: "thermostatSetpointRange", value: thermostatSetpointRange, displayed: false)
-	sendEvent(name: "heatingSetpointRange", value: heatingSetpointRange, displayed: false)
+	configureSupportedRanges()
 
+	unschedule(scheduledUpdateWeather)
 	if (settings.zipcode) {
-		requests += scheduledUpdateWeather()
-		unschedule(scheduledUpdateWeather)
+		requests += updateWeather()
```

Comment:
updateWeather returns an object we can add to the list to send, scheduledUpdateWeather actually sends the object and returns nothing, so use updateWeather